include/firrtl.xml

```diff
@@ -152,26 +152,26 @@
       </context>
       <context name="outertype" attribute="ID" lineEndContext="#pop">
         <DetectChar char="&lt;" attribute="Operator" context="#stay"/>
+        <keyword String="outertype" attribute="Keyword" context="outertype"/>
         <keyword String="types" attribute="Keyword" context="type"/>
         <Detect2Chars char="{" char1="|" context="field" attribute="Separator"/>
-        <Detect2Chars char="|" char1="}" attribute="Separator" context="#pop#pop"/>
         <DetectChar char="{" context="field" attribute="Separator"/>
-        <DetectChar char="}" attribute="Separator" context="#pop#pop"/>
         <DetectChar char= ";" context="comment" attribute="Comment"/>
-        <AnyChar String="&gt;," attribute="Operator" context="#pop"/>
-        <DetectChar char="[" attribute="Operator" context="widthOrDepthOrLit"/>
+        <DetectChar char="&gt;" attribute="Operator" context="#pop"/>
+        <DetectChar char="[" attribute="Operator" context="widthOrDepth"/>
+        <DetectChar char="," attribute="Separator" context="#stay"/>
       </context>
       <context name="type" attribute="String" lineEndContext="#pop">
-        <AnyChar String="&lt;[(" attribute="Operator" context="widthOrDepthOrLit"/>
+        <AnyChar String="&lt;[" attribute="Operator" context="widthOrDepth"/>
+        <DetectChar char="(" attribute="Separator" context="#pop"/>
         <AnyChar String="," attribute="Separator" context="#pop"/>
-        <AnyChar String="}" attribute="Separator" context="#pop#pop"/>
-        <Detect2Chars char="|" char1="}" attribute="Separator" context="#pop#pop"/>
+        <AnyChar String="}" attribute="Separator" context="#pop" lookAhead="true"/>
```

Comment:
If in the `type` context, it's because we saw, e.g., `UInt`.  It optionally followed by width/depth.  Any paren indicates the type portion is over (don't treat `UInt<5>(..` as in the "type" context).

Use lookAhead to exit the type context without consuming the character, for when a bundle or enum or outertype contains a type.